test/utils.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Checks that each of the values of two array-like objects are deep equal.
+ * This is needed because jasmine's toEqual compares the functional
+ * attributes of dataLayer, which includes a push function.
+ * Since the push function is an anonymous function that gets
+ * recreated every time, we can't create an object that
+ * is deep equal to any dataLayer.
+ * @param {!Array<*>} arr1 The first array-like object
+ * @param {!Array<*>} arr2 The second array-like object
+ */
+function expectEqualContents(arr1, arr2) {
```

Comment:
For a moment I was confused why `expect(arr1).toEqual(arr2)` was not sufficient here.  After reading the JSDoc though that was clear.

I wonder if the function name and arguments could clarify it a bit.  WDYT about renaming this to expectDataLayerEquals(expected, dataLayer)?

Reply:
Ok, I changed the name.